strongbox-security/strongbox-authentication-registry/src/main/java/org/carlspring/strongbox/authentication/ConfigurableProviderManager.java

```diff
@@ -104,54 +102,49 @@ private void reloadAuthenticationItems()
     public UserDetails loadUserByUsername(String username)
         throws UsernameNotFoundException
     {
-        UserData user = loadUserDetails(username);
+        Optional<User> user = loadUserDetails(username);
 
-        if (user == null)
+        if (!user.isPresent())
         {
             throw new UsernameNotFoundException(username);
         }
 
-        return springSecurityUserMapper.apply(user);
+        return user.map(userDetailsMapper).get();
     }
 
-    private UserData loadUserDetails(String username)
+    private Optional<User> loadUserDetails(String username)
     {
-        UserData user = userService.findByUserName(username);
+        User user = strongboxUserManager.findByUserName(username);
 
         Date userLastUpdate = Optional.ofNullable(user)
                                       .flatMap(u -> Optional.ofNullable(u.getLastUpdate()))
                                       .orElse(Date.from(Instant.EPOCH));
         Date userExpireDate = Date.from(Instant.ofEpochMilli(userLastUpdate.getTime())
-                                               .plusSeconds(USER_INVALIDATE_SECONDS));
+                                               .plusSeconds(externalUsersInvalidateSeconds));
         Date nowDate = new Date();
-        if (user != null && nowDate.before(userExpireDate))
+        if (user != null && (StringUtils.isBlank(user.getSourceId()) || nowDate.before(userExpireDate)))
         {
-            return user;
+            return Optional.of(user);
         }
-
-        UserDetails externalUserDetails = null;
-        for (UserDetailsService userDetailsService : userProviderMap.values())
+       
+        for (Entry<String, UserDetailsService> userDetailsServiceEntry : userProviderMap.entrySet())
         {
+            String sourceId = userDetailsServiceEntry.getKey();
+            UserDetailsService userDetailsService = userDetailsServiceEntry.getValue();
             try
             {
-                externalUserDetails = userDetailsService.loadUserByUsername(username);
-                
-                break;
+                return Optional.of(strongboxUserManager.updateExternalUserDetails(sourceId,
+                                                                                  userDetailsService.loadUserByUsername(username)));
             }
             catch (UsernameNotFoundException e)
             {
                 continue;
             }
         }
+        
+        strongboxUserManager.deleteByUserName(username);
```

Comment:
Cosmetic: This method's name should be `deleteByUsername(...)`.

Reply:
done

Reply:
Thanks! :)